pkg/utils/resources/resources.go

```diff
@@ -63,6 +63,23 @@ func Merge(resources ...v1.ResourceList) v1.ResourceList {
 	return result
 }
 
+// MergeResourceLimitsIntoRequests merges resource limits into requests if no request exists for the given resource
+func MergeResourceLimitsIntoRequests(container v1.Container) v1.ResourceList {
+	if container.Resources.Requests == nil {
+		container.Resources.Requests = v1.ResourceList{}
+	}
+	if container.Resources.Limits == nil {
+		return container.Resources.Requests
+	}
+	resource := container.Resources.Requests
```

Comment:
I don't think we want to modify the resource list on the container here.  The client we use has a cache, so those modifications will be seen by us later.  It's safer (but slower) to not modify it and just return a new v1.ResourceList that has the correct values.

Reply:
Good call.  I'll take a look at that.